src/adapter/src/coord/command_handler.rs

```diff
@@ -774,10 +785,17 @@ impl Coordinator {
         }
     }
 
+    /// Chooses a timestamp for `mz_now()`, if `mz_now()` occurs in the `with_options` of the materialized view.
+    /// If `acquire_read_holds` is true, it also grabs read holds on input collections that might possibly be involved
+    /// in the MV.
+    ///
+    /// Note that this is NOT what handles `mz_now()` in the query part of the MV. (handles it only in `with_options`).
     async fn resolve_mz_now_for_create_materialized_view(
-        &self,
+        &mut self,
         cmvs: &CreateMaterializedViewStatement<Aug>,
         resolved_ids: &ResolvedIds,
+        acquire_read_holds: bool,
```

Comment:
Instead of a `bool` function argument, what about `acquire_read_holds_for: Option<&mut Session>`? That way the mut session is only passed if needed, and also serves as the flag to auto cleanup the read hold?

Reply:
Good idea!